cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1403,8 +1536,31 @@ pub(crate) fn emit(
             };
 
             let dst = &dst.finalize(state);
+            if let Some(srcloc) = *srcloc {
+                // Register the offset at which the actual load instruction starts.
+                sink.add_trap(srcloc, TrapCode::HeapOutOfBounds);
+            }
             emit_std_reg_mem(sink, prefix, opcode, 2, *src, dst, rex);
         }
+
+        Inst::LoadExtName {
+            dst,
+            name,
+            offset,
+            srcloc,
+        } => {
+            // The full address can be encoded in the register, with a relocation.
```

Comment:
Add a comment giving the instruction this is emitting? `movq $imm64, %dst` or similar...

Reply:
+1 for that; please add comments always, indicating what insns are being emitted.  Inferring from the opcodes is nearly impossible.

Reply:
Also .. is there a typo in the commit message?  "movaps" isn't the insn you mean AFAIK; that's an aligned SSE load (move-aligned-packed-singleprecision).  Maybe you mean "movabsq" ?